enhancements/edge-topologies/adaptable-topology.md

```diff
@@ -0,0 +1,1078 @@
+- Early coordination with the Bare Metal Networking team
+- Alternative is to document the limitation
+- `platform: none` provides full support as a fallback
+
+### Drawbacks
+
+#### Increased Operator Complexity
+
+Operators must be updated to handle dynamic topology changes based on node counts.
+The logic for different node counts already exists in operators.
+It's currently applied once at startup because Infrastructure config
+topology values are static.
+Operators must now watch for node count changes and adjust behavior dynamically.
+Library-go utilities help, but operators still need to test and validate
+behavior across all node count thresholds.
+
+#### Expanded Test Matrix
+
+Testing Adaptable topology requires validating behavior at each node count
+threshold (1, 2, 3+ control-plane nodes; 0, 1, 2+ worker nodes).
+This expands the test matrix compared to testing a single fixed topology.
+CI infrastructure must handle additional test lanes for installations,
+transitions, and scaling operations.
+
+#### One-Way Transitions
+
+Transitions to Adaptable topology are one-way.
+Clusters cannot transition back to fixed topology modes.
+This design simplifies implementation but means users must be certain
+before transitioning.
+Mistakes require cluster redeployment.
+
+#### Coordination Across Teams
+
+Adaptable topology requires updates across many teams:
+installer, oc CLI, library-go, CEO, OLM, Console, and all core operator teams.
+This coordination overhead is significant.
+Any team falling behind delays the entire feature.
+
+#### Limited Optional and 3rd Party Operator Support
+
+Optional and 3rd party operators have no required timeline for
+Adaptable topology support updates.
+Users may encounter operators that don't handle topology transitions correctly.
+Operator compatibility annotations help identify these cases,
+but users may still need to wait for updates or choose alternative operators.
+
+#### Potential for Unexpected Behavior
+
+Users accustomed to fixed topologies may encounter unexpected behavior
+when node counts cross thresholds.
+The behavior is deterministic and documented, but differs from fixed topologies.
+Troubleshooting may be more complex when behavior changes based on node count.
+
+## Alternatives (Not Implemented)
+
+### Direct Topology Transitions
+
+An alternative is to allow direct transitions between topology modes
+(e.g., SingleReplica → HighlyAvailable) without introducing Adaptable topology.
+
+Transitioning from SingleReplica to HighlyAvailable signals the user wants
+HA cluster behavior.
+
+However, several problems emerge:
+
+**Ambiguous Target State**: Should transitioning to HighlyAvailable create a
+3-node compact cluster or provision compute nodes for a 5-node cluster?
+The end state is unclear.
+
+**Topology Mismatch**: The cluster configuration would claim HighlyAvailable
+while still operating as SingleReplica until nodes are provisioned.
+The declared topology differs from actual behavior.
```

Comment:
Adaptable will have this intermediate state, too. E.g. during etcd learner phase when transitioning from SNO to MNO